### ISSUE TITLE
Wrap site-settings page in auth component

### ISSUE THREAD

**Issue by crookedneighbor** [Commander-Spellbook/website-v2#226]
Blocked by https://github.com/Commander-Spellbook/website-v2/pull/213 and https://github.com/Commander-Spellbook/website-v2/issues/225

Right now, the site-settings link only shows up if you have the permission, but nothing stops a user from navigating to the page directly. (the API route still restricts changes)

Once the permission check component is completed, we can wrap this page in it.

Acceptance criteria:
- [ ] wrap site-settings with the `manageSiteContent` permission

**Comment by crookedneighbor** [comment]
No longer relevant as the backend system is happening in a separate app now.
